app/src/main/java/com/hedvig/app/feature/marketpicker/LanguagePickerBottomSheet.kt

```diff
@@ -0,0 +1,47 @@
+package com.hedvig.app.feature.marketpicker
+
+import android.content.DialogInterface
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.lifecycle.observe
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import com.hedvig.app.R
+import com.hedvig.app.databinding.LanguagePickerBottomSheetBinding
+import com.hedvig.app.feature.settings.Language
+import com.hedvig.app.util.extensions.viewBinding
+
+class LanguagePickerBottomSheet(
+    private val viewModel: MarketPickerViewModel,
+    private val tracker: MarketPickerTracker
+) : BottomSheetDialogFragment() {
+    val binding by viewBinding(LanguagePickerBottomSheetBinding::bind)
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? = inflater.inflate(R.layout.language_picker_bottom_sheet, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        binding.apply {
+            recycler.adapter = LanguagePickerBottomSheetAdapter(viewModel, tracker, dialog)
+            viewModel.data.observe(viewLifecycleOwner) { VMState ->
+                VMState.market?.let { market ->
+                    (recycler.adapter as LanguagePickerBottomSheetAdapter).items =
+                        Language.getAvailableLanguages(market)
+                }
+                (recycler.adapter as LanguagePickerBottomSheetAdapter).notifyDataSetChanged()
+            }
+        }
+    }
+
+    override fun onCancel(dialog: DialogInterface) {
+        viewModel.save()
```

Comment:
Yes, when a market is selected, we call `dialog?.cancle()` which then closes the bottomsheet and saves if it is not the same language and market

Reply:
Could we not avoid the call to `MarketPickerViewModel::save` entirely if we do not call that function from `LanguagePickerBottomSheet::onCancel` and instead call it directly from the click listener, and then direct the dialog to be canceled?

Reply:
I get some weird behaviours if i do it in the click listener since we reload the activity and also the `LanguagePickerBottomSheet` and it works as intended if i do it this way